### PR TITLE
bump cbmc dependency

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -134,7 +134,7 @@ jobs:
       - name: Get minisat
         run: make -C lib/cbmc/src minisat2-download
       - name: Build with make
-        run: make CXX="ccache g++" -C src -j2
+        run: make CXX="ccache g++ -lstdc++fs -Wno-class-memaccess" -C src -j2
```
Comment (tautschnig) [inline .github/workflows/pull-request-checks.yaml]: I believe this needs to be `make CXX="ccache g++ -Wno-class-memaccess" LIBS="-lstdc++fs" -C src -j2` to work.

File: .github/workflows/pull-request-checks.yaml [hunk 2]
```diff
@@ -44,8 +44,8 @@ jobs:
         run: make -C src -j2 CXX="ccache g++"
       - name: Run the ebmc tests with SAT
         run: make -C regression/ebmc test
-      - name: Run the ebmc tests with Z3
-        run: make -C regression/ebmc test-z3
+      #- name: Run the ebmc tests with Z3
+      #  run: make -C regression/ebmc test-z3
```
Comment (tautschnig) [inline .github/workflows/pull-request-checks.yaml:48]: Should these really be disabled?

File: .github/workflows/pull-request-checks.yaml [hunk 3]
```diff
@@ -94,8 +94,8 @@ jobs:
           make CXX="ccache clang++" -C src -j2
       - name: Run the ebmc tests with SAT
         run: make -C regression/ebmc test
-      - name: Run the ebmc tests with Z3
-        run: make -C regression/ebmc test-z3
+      #- name: Run the ebmc tests with Z3
+      #  run: make -C regression/ebmc test-z3
```
Comment (tautschnig) [inline .github/workflows/pull-request-checks.yaml:98]: Should these really be disabled?

**Review by tautschnig — APPROVED** [review]

File: .github/workflows/pull-request-checks.yaml (hunk 2 above)
Comment (kroening) [reply .github/workflows/pull-request-checks.yaml:48]: This needs https://github.com/diffblue/cbmc/pull/8123.
I won't wait for that, will merge #315.

**Comment by kroening** [comment]
The Z3 tests are back; this is now bumping to the commit of https://github.com/diffblue/cbmc/pull/8123.

**Pull request merged**